src/streams/register_nack.rs

```diff
@@ -0,0 +1,560 @@
+use std::ops::Range;
+
+use crate::rtp_::{Nack, NackEntry, ReportList, SeqNo};
+
+/// How many sequence numbers we keep track of
+const MAX_DROPOUT: u64 = 3000;
+
+/// Number of out of order packets we keep track of for reports
+/// A suggested maximum value min is MAX_DROPOUT / 2
+const MAX_MISORDER: u64 = 100;
+
+const U16_MAX: u64 = u16::MAX as u64 + 1_u64;
+
+/// The max number of NACKs we perform for a single packet
+const MAX_NACKS: u8 = 5;
+
+#[derive(Debug)]
+pub struct NackRegister {
+    /// Status of packets indexed by wrapping SeqNo.
+    packets: Vec<PacketStatus>,
+
+    /// Range of seq numbers considered NACK reporting.
+    active: Option<Range<SeqNo>>,
+}
+
+#[derive(Debug, Default, Clone, Copy)]
+struct PacketStatus {
```

Comment:
Nitpic: This ends up being 2 bytes in size because of the `received` `bool`. We could cut that in half by storing the received flag in the most significant bit. So instead of being 6KB in memory it would be 3KB.

Reply:
Maybe make it work before optimizing like that?

Will make it harder to read.